### PR TITLE
CI: Use run_task from maintainer tools

### PR DESCRIPTION
**Pull request by tcharding**
This PR slightly improves test coverage by linting the integration tests, no meaningful change of coverage really though just improvement to the CI setup.

### CODE REVIEW

**Comment by tcharding** [comment]
Lint warnings are real, they are in the previously un-linted `bitcoind-tests`.

**Comment by storopoli** [comment]
No love (`README.md`) for the `miniscript` crate?

**Comment by tcharding** [comment]
Includes `.github/workflows/README.md` as requested. These files risk going stale though.

**Comment by storopoli** [comment]
> Includes `.github/workflows/README.md` as requested. These files risk going stale though.

Yes, but stale documentation always reveals more than non-existent.

**Comment by tcharding** [comment]
Fair point.

**Comment by tcharding** [comment]
This will need to be rebased on #685

**Comment by tcharding** [comment]
Ready to go!

File: bitcoind-tests/tests/setup/test_util.rs [hunk 1]
```diff
@@ -86,8 +86,8 @@ fn setup_keys(
     let mut x_only_keypairs = vec![];
     let mut x_only_pks = vec![];
 
-    for i in 0..n {
-        let keypair = bitcoin::secp256k1::Keypair::from_secret_key(&secp_sign, &sks[i]);
+    for sk in sks.iter().take(n) {
```
Comment (apoelstra) [inline bitcoind-tests/tests/setup/test_util.rs:89]: In 3ad8c0e9615d6e9369a3ae37d66f4697d205ca33:

Pretty sure `take` is unnecessary here. You can just do `for sk in &sks`.

**Comment by apoelstra** [comment]
Other than the single nit befd49cb5cc62af63f28abba3bb4dff9a09fa8e5 looks great!

Can you please split this into two PRs -- doing the clippy stuff as one and the CI stuff as another (starting with the "add lockfiles" commit) would be a more natural split. Plus it would be really annoying/difficult for me to locally test a PR where the lockfile location were to change partway through.

Finally, what happens to the existing contrib/test.sh? Should there be another commit which deletes it?

**Comment by tcharding** [comment]
Thanks man, I'll sort it out!

**Comment by apoelstra** [comment]
>Plus it would be really annoying/difficult for me to locally test a PR where the lockfile location were to change partway through.

Just so you know this sort of thing is no longer annoying or difficult for me. My CI script is now smart enough to look for lockfiles in several locations within the source tree, and to fallback to externally provided lockfiles if none are present.

**Comment by tcharding** [comment]
Sweet, nice one.

**Comment by apoelstra** [comment]
In 46d59b7cf122def8d4be77f4ab7de62d741846b7 can you fold the lockfile changes into the first commit so that the lockfiles are correct as soon as they appear?

As for "the lockfile-updating script should appear in maintainer-tools", I'm tempted to move to a model where we copy the files out of rust-bitcoin-maintainer-tools, along with a file indicating their git commit of origin, and in CI we check that the files are exact copies. Then people have all the scripts available locally without needing to go to the other repo, but we still guarantee everything is in sync.

But that doesn't need to be part of this PR.

**Comment by tcharding** [comment]
Put all the lock file stuff in a single patch and put the clippy fix up front. No other changes.

**Review by apoelstra — APPROVED** [review]
ACK 6fffb8bb0f74cf9d636b34e9afc8227a1804c5d7

**Review by ErnestoLara — APPROVED** [review]

**Review by ErnestoLara — APPROVED** [review]

**Review by ErnestoLara — APPROVED** [review]

**Review by ErnestoLara — APPROVED** [review]

**Pull request merged**